devito/nodes.py

```diff
@@ -350,18 +358,28 @@ def ccode(self):
 
         # For reverse dimensions flip loop bounds
         if self.reverse:
-            loop_init = c.InlineInitializer(c.Value("int", self.index),
-                                            ccode('%s - 1' % end))
+            loop_init = 'int %s = %s' % (self.index, ccode('%s - 1' % end))
             loop_cond = '%s >= %s' % (self.index, ccode(start))
             loop_inc = '%s -= %s' % (self.index, self.limits[2])
         else:
-            loop_init = c.InlineInitializer(c.Value("int", self.index), ccode(start))
+            loop_init = 'int %s = %s' % (self.index, ccode(start))
             loop_cond = '%s < %s' % (self.index, ccode(end))
             loop_inc = '%s += %s' % (self.index, self.limits[2])
 
+        # Append unbounded indices, if any
+        if self.uindices:
```

Comment:
Unbounded indices (see also the docstrings of `Iteration`  and `UnboundedIndex`) are special indices that "belong" to an iteration, but are used to traverse a Dimension in a non-linear fashion; that is, instead of increasing in ``+= 1`` steps, their value can change at each iteration in any way (e.g., using ``%``, ``+= k``, etc).

An UnboundedIndex is still related to an Iteration though, because for example it derives from a ``BufferedDimension`` or any future (subclass of) ``Dimension`` object that we plan to introduce; hence, they're now generated in the loop header directly, instead of "floating at random places within the code".

Example:
Previously we had something like
```
for (t = 0; t < N; t++) {
  t1 = t0 % 2;
}
```

Now instead we generate:
```
for (t = 0, t1 = t0 % 2; t < N; t++, t1 = t0 % 2) {
}
```

To make it simple, this simple change makes compiler analysis simpler and faster. Functionally, there is no difference.